src/basic/log.h

```diff
@@ -191,9 +191,10 @@ void log_assert_failed_return(
 #define log_full_errno(level, error, ...)                               \
         ({                                                              \
                 int _level = (level), _e = (error);                     \
-                (log_get_max_level() >= LOG_PRI(_level))                \
+                _e = (log_get_max_level() >= LOG_PRI(_level))           \
                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
                         : -ERRNO_VALUE(_e);                             \
+                _e < 0 ? _e : -EIO;                                     \
```

Comment:
wouldn't EINVAL be a better generic "je ne sais quoi" error?

Reply:
This is a sign of a programming error (hopefully not in our code). We often use EIO in this fashion. It might even be better to use something *less* common, e.g. ESTRPIPE? (Errno is not directly connected to streams, but both are obsolete concepts from long ago… so it's fitting to use this code.)

Reply:
ENONRECOVERABLE